src/nvim/os/shell.c

```diff
@@ -80,21 +80,49 @@ char **shell_build_argv(const char *cmd, const char *extra_args)
 void shell_free_argv(char **argv)
 {
   char **p = argv;
-
   if (p == NULL) {
     // Nothing was allocated, return
     return;
   }
-
   while (*p != NULL) {
     // Free each argument
     xfree(*p);
     p++;
   }
-
   xfree(argv);
 }
 
+/// Joins shell arguments from `argv` into a new string.
+///
+/// @returns[allocated] `argv` joined to a string.
+char *shell_argv_to_str(char **const argv)
+  FUNC_ATTR_NONNULL_ALL
+{
+  size_t n = 0;
+  char **p = argv;
+  if (p == NULL) {
+    return 0;
+  }
+
+  const size_t maxsize = 256;
+  char *rv = xcalloc(maxsize, sizeof(*rv));
+  while (*p != NULL && n < maxsize) {
+    n += xstrlcat(rv, "'", maxsize);
```

Comment:
`xstrlcat()` returns the untruncated length of the new string `rv`, not the length of the second string. So 
`n = xstrlcat(...);` should be used.

Reply:
Perhaps `xstrlcpy()` could be used here instead, so that only the length of the second argument is computed and added?